### ISSUE TITLE
tox4: cannot handle empty envlist for `tox -l`

### ISSUE THREAD

**Issue by jugmac00** [tox-dev/tox#1987]
```
❯ tox4 -l
Traceback (most recent call last):
  File "/home/jugmac00/.local/bin/tox4", line 8, in <module>
    sys.exit(run())
  File "/home/jugmac00/.local/lib/python3.6/site-packages/tox/run.py", line 22, in run
    result = main(sys.argv[1:] if args is None else args)
  File "/home/jugmac00/.local/lib/python3.6/site-packages/tox/run.py", line 47, in main
    result = handler(state)
  File "/home/jugmac00/.local/lib/python3.6/site-packages/tox/session/cmd/legacy.py", line 98, in legacy
    return list_env(state)
  File "/home/jugmac00/.local/lib/python3.6/site-packages/tox/session/cmd/list_env.py", line 27, in list_env
    max_length = max(len(env) for env in (default.envs + extra))
ValueError: max() arg is an empty sequence
```

## reproduction

Just delete the complete content of `tox.ini` on the rewrite branch.

(I noticed this problem when trying to reproduce #1985)

Interestingly (for me), `tox4 list` does not crash... 

TIL: `tox -l` and `tox list` are not aliases => `-l` and `--listenvs` are legacy options, and not the same as `list`

```
❯ tox4 list
additional environments:
py -> [no description]
```
(btw what is `py`??? and why is there an entry, when there is no additional environment)

**Comment by jugmac00** [comment]
The issue looks straightforward (except for the `additional environment: py`).

I'd like to work on this one.

**Comment by gaborbernat** [comment]
Because empty envlist is translated to ``py`` extra. That's by design.

**Comment by jugmac00** [comment]
closed by #1988
